cirrus/server/cirrus/main.py

```diff
@@ -27,5 +44,39 @@ async def compute_enrollment_status():
     client_feature_configuration = fml.compute_feature_configurations(
         enrolled_partial_configuration, feature_configurations
     )
-
     return client_feature_configuration
+
+
+@app.on_event("startup")
+async def start_scheduler():
+    scheduler.start()
+    app.state.scheduler = scheduler
+
+
+@app.on_event("shutdown")
+async def shutdown_scheduler():
+    scheduler.shutdown()
+
+
+async def fetch_schedule_recipes() -> None:
+    try:
+        remote_setting.fetch_recipes()
+    except Exception as e:
+        # If an exception is raised, log the error and schedule a retry
+        logger.error(f"Failed to fetch recipes: {e}")
+        scheduler.add_job(  # type: ignore
+            fetch_schedule_recipes,
+            "interval",
+            seconds=remote_setting_refresh_rate_in_seconds,
+            max_instances=1,
+            max_retries=3,
+        )
+
+
+# Schedule the initial job to fetch recipes
+scheduler.add_job(  # type: ignore
+    fetch_schedule_recipes,
+    "interval",
```

Comment:
Something I think we need to discuss at some point is intervals vs cron, and managing memory state across the containers. 

If all containers' intervals aren't lined up, then some might have more recent/older recipes than others. Similarly with cron though, while I'd argue we _can_ trust that all containers will have the same system time, if containers go down and they fetch recipes when they come online, we could still end up in that state albeit everything will re-sync at the next cron match.

Maybe it's not a huge concern, I just want to note that I think it is at a minimum a valid concern we should provide options for mitigating. Maybe it's something we work out in the production-hardening phase.

Reply:
yes I also thought about that, I think eventually they will get updated copy as they are planning to send request to cirrus with every backend call 🤔

Reply:
Are the recipes going to be fetched on every call to the enrollment endpoint?

The concern I was trying to share is if two different services ping two different cirrus pods as a part of a single request flow, each pod could have two different sets of recipes because the intervals could be triggering at different times.

Reply:
they will be fetched in the background, so whatever current copy of recipe it will have in the memory will use that, but I think in single request flow it will only be hitting one cirrus pod per request, isn't it? I agree two pod may have different copy, it also depends how often we get new recipes

Reply:
I'm not clear on their plans for deployment.

It's definitely something we can do in the production hardening phase, I just wanted to go ahead and note it.

Reply:
makes sense, I have noted down this we can circle back :)

Reply:
Yeah that's a super interesting point @jeddai becuase if you have N copies of the container and their RS data are out of sync, they'll give back different possibly conflicting feature configs throughout the consuming applications architecture.  Suuuuuper interesting.  Let's think about that some more!

Reply:
they can have same copy if we schedule all jobs in sync, like crontab

Reply:
Yep and `appscheduler` can use cron instead of an interval, so I really think it should be a configurable thing. 

If we want to get into making sure this _can't_ happen then we'll likely need to talk about some service discoverability stuff if we want to keep it ephemeral. But again all this can totally be figured out in the hardening phase, just wanted to go on record with it now